### PR TITLE
OA: interface update

### PR DESCRIPTION
**Pull request by mrivi**
- Add the waypoint type (mavlink command associated with the waypoint) to the points in the OA interface
- refactor the check on the z progress of the mission

### CODE REVIEW

**Comment by dagar** [comment]
Did you see the CI failures? http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-12349/1/pipeline

``` Console
../../src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp:55:1: error: narrowing conversion of '255' from 'int' to 'bool' inside { } [-Werror=narrowing]

 };

 ^
```

**Comment by dagar** [comment]
SITL MC avoidance as well.

http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-SITL_tests/detail/PR-12349/1/pipeline

**Comment by mrivi** [comment]
@dagar We cannot merge this one until there is a mavros update and consequent CI update. Problem is that `TRAJECTORY_REPRESENTATION_WAYPOINT` changed so the mavlink receiver is dropping all the messages coming from ROS because the definition don't match. It isn't an extention of the message but a proper change since it's still marked WIP in mavlink.

Also updating the mavlink submodule on its own will break the avoidance CI.

File: src/modules/mc_pos_control/CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ px4_add_module(
 	MODULE modules__mc_pos_control
 	MAIN mc_pos_control
 	COMPILE_FLAGS
-	STACK_MAIN 1300
+	STACK_MAIN 1500
```
Comment (bkueng) [inline src/modules/mc_pos_control/CMakeLists.txt]: Why is this needed?
